src/Uuid/ModelUuid.php

```diff
@@ -69,15 +69,31 @@ public function id(): string
 		$user  = $kirby->auth()->currentUserFromImpersonation();
 		$kirby->impersonate('kirby');
 
+		// needed for multi-language setup
+		$languageCode = null;
+		$isMultilang = $kirby->multilang() === true &&
+			$kirby->languageCode() !== $kirby->defaultLanguage()->code();
+
+		// if multilang enabled and current language is not default
+		// get UUID from the default language content
+		if ($isMultilang === true) {
+			$languageCode = $kirby->defaultLanguage()->code();
+		}
+
 		// get the content array from the page
-		$data = $this->model->content()->toArray();
+		$data = $this->model->content($languageCode)->toArray();
 
 		// check for an empty content array
 		// and read content from file again,
 		// just to be sure we don't lose content
 		if (empty($data) === true) {
 			usleep(1000);
-			$data = $this->model->readContent();
+			$data = $this->model->readContent($languageCode);
+		}
+
+		// return UUID from the default language content if available
+		if ($isMultilang === true && empty($data['uuid']) === false) {
+			return $data['uuid'];
```

Comment:
Shouldn't this already be covered by `Uuid::__construct()`/`ModelUuid::retrieveId()`? Kirby has an automatic fallback to the default language content already.

Reply:
@lukasbestle Ahh, nice catch! If the secondary language is called, calling `::retrieveId()` will already run `->id()` for the primary and the UUID will be generated and saved for primary. Did I understand the process correctly?

Reply:
> To cover the third, I think we should modify ModelUuid::retrieveId() to always access the default language content.

I've tested and seems works great. Do you think still need this one?

Reply:
> If the secondary language is requested, calling `::retrieveId()` will already run `->id()` for the primary and the UUID will be generated and saved for primary. Did I understand the process correctly?

Yep. But that only works if there is no different UUID stored in the secondary language content file...

> Do you think still need this one?

...so I think we need the change in `retrieveId()`. The UUID cannot end up in secondary language content files programmatically with your change in `saveTranslation()`, but it could still end up there manually.

Reply:
PS: The call to `->id()` only happens in `Uuid::retrieveId()`, but this method is overridden in `ModelUuid::retrieveId()` with a retrieval from content. I think we only need to pass the default language code there.

Reply:
I couldn't figure out the CI issue about kirbytags: https://github.com/getkirby/kirby/actions/runs/3311545395/jobs/5467126691

Reply:
The change looks good to me.

I've fixed the unit test. The failure was caused by the UUID being stored in the translated content.